api/checks/webhook.go

```diff
@@ -108,15 +108,17 @@ func handleCheckRunEvent(ctx context.Context, payload []byte) (bool, error) {
 		return false, err
 	}
 
-	if checkRun.Action != nil &&
-		(*checkRun.Action == "created" || *checkRun.Action == "rerequested") {
-		name, sha := *checkRun.CheckRun.Name, *checkRun.CheckRun.HeadSHA
-		log.Debugf("Check run %s @ %s %s", name, sha[:7], *checkRun.Action)
-		spec, err := shared.ParseProductSpec(*checkRun.CheckRun.Name)
-		if err != nil {
-			log.Errorf("Failed to parse \"%s\" as product spec")
+	if checkRun.Action != nil {
```

Comment:
Nit: we have too many levels of nesting here and this one feels like an error check, so let's convert this top-level if into an early return.